Cache/Hashing.lean

```diff
@@ -11,8 +11,32 @@ namespace Cache.Hashing
 
 open System IO
 
-/-- We cache the hash of each file for faster lookup -/
-abbrev HashM := StateT IO.HashMap IO
+structure HashMemo where
+  depsMap : Lean.HashMap FilePath (Array FilePath)
+  hashMap : HashMap
+  deriving Inhabited
+
+/-- Tells whether a string matches a pattern with at most one wildcard '*' -/
+def matchesAnyPattern (str : String) : List String → IO Bool
+  | [] => pure false
+  | p :: ps => match p.splitOn "*" with
```

Comment:
Can we please use a different character than `*`?  As Eric already mentioned, this will usually be expanded by unix shells.  And if it's not expanded then it will just fail in sensible shells:
```
$ lake exe cache get *Conjugate.lean
fish: No matches for wildcard “*Conjugate.lean”. See `help expand`.
lake exe cache get *Conjugate.lean
                   ^
```

Reply:
Can we use `%` then?